hibernate-core/src/main/java/org/hibernate/cfg/annotations/BasicValueBinder.java

```diff
@@ -239,7 +255,23 @@ public int getPreferredSqlTypeCodeForArray() {
 
 	@Override
 	public boolean isNationalized() {
-		return isNationalized;
+		if ( isNationalized ) {
+			return true;
+		}
+		if ( explicitJdbcTypeAccess != null ) {
+			final JdbcType type = explicitJdbcTypeAccess.apply( getTypeConfiguration() );
+			if ( type != null ) {
+				switch ( type.getJdbcTypeCode() ) {
+					case SqlTypes.NCHAR:
+					case SqlTypes.NVARCHAR:
+					case SqlTypes.LONGNVARCHAR:
+					case SqlTypes.LONG32NVARCHAR:
+					case SqlTypes.NCLOB:
+						return true;
+				}
```

Comment:
Following the suggestion above, how about adding `JdbcType#isNationalized` and using that here?

```suggestion
				return type.isNationalized();
```